cmd/network-passt-binding/domain/configurator.go

```diff
@@ -0,0 +1,225 @@
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package domain
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	vmschema "kubevirt.io/api/core/v1"
+
+	domainschema "kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+
+	"kubevirt.io/client-go/log"
+
+	"kubevirt.io/kubevirt/pkg/network/istio"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/device"
+
+	"kubevirt.io/kubevirt/pkg/network/namescheme"
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+)
+
+type NetworkConfiguratorOptions struct {
+	IstioProxyInjectionEnabled bool
+	UseVirtioTransitional      bool
+	UseLaunchSecurity          bool
+	LogFile                    string
+	VirtioNetProhibited        bool
+	QueueCount                 uint
+}
+
+type PasstNetworkConfigurator struct {
+	vmiSpecIface *vmschema.Interface
+	options      NetworkConfiguratorOptions
+}
+
+// PasstPluginName passt binding plugin name should be registered to Kubevirt through Kubevirt CR
+const PasstPluginName = "passt"
+
+func NewPasstNetworkConfigurator(ifaces []vmschema.Interface, networks []vmschema.Network, opts NetworkConfiguratorOptions) (*PasstNetworkConfigurator, error) {
+	network := vmispec.LookupPodNetwork(networks)
+	if network == nil {
+		return nil, fmt.Errorf("pod network not found")
+	}
+	iface := vmispec.LookupInterfaceByName(ifaces, network.Name)
+	if iface == nil {
+		return nil, fmt.Errorf("no interface found")
+	}
+	if iface.Binding == nil && iface.Passt == nil {
+		return nil, fmt.Errorf("interface %q is not set with Passt network binding plugin or Passt binding method", network.Name)
+	}
+	if iface.Binding != nil && iface.Binding.Name != PasstPluginName {
+		return nil, fmt.Errorf("interface %q is not set with Passt network binding plugin", network.Name)
+	}
+
+	return &PasstNetworkConfigurator{
+		vmiSpecIface: iface,
+		options:      opts,
+	}, nil
+}
+
+func (s PasstNetworkConfigurator) Mutate(domainSpec *domainschema.DomainSpec) (*domainschema.DomainSpec, error) {
+	generatedIface, err := s.generateInterface()
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate domain interface spec: %v", err)
+	}
+
+	domainSpecCopy := domainSpec.DeepCopy()
+	if index, iface := lookupIfaceByAliasName(domainSpec.Devices.Interfaces, s.vmiSpecIface.Name); iface != nil {
+		if !reflect.DeepEqual(iface, generatedIface) {
+			domainSpecCopy.Devices.Interfaces[index] = *generatedIface
+		}
+	} else {
+		domainSpecCopy.Devices.Interfaces = append(domainSpecCopy.Devices.Interfaces, *generatedIface)
+	}
+
+	return domainSpecCopy, nil
+}
+
+func lookupIfaceByAliasName(ifaces []domainschema.Interface, name string) (int, *domainschema.Interface) {
+	for i, iface := range ifaces {
+		if iface.Alias != nil && iface.Alias.GetName() == name {
+			return i, &ifaces[i]
+		}
+	}
+
+	return -1, nil
+}
+
+func (s PasstNetworkConfigurator) generateInterface() (*domainschema.Interface, error) {
+	if s.options.VirtioNetProhibited && (s.vmiSpecIface.Model == vmschema.VirtIO || s.vmiSpecIface.Model == "") {
```

Comment:
Can this be moved to a dedicated function that expresses what it means in the name? (including the error text)

Reply:
This check is now removed

Reply:
This drop is not recorded in the commit message and it seems to be significant.
I am having hard time understanding if this is indeed redundant or not. I do not see how it is assured that the resource has been assigned.

Are you saying that the pod was not suppose to be scheduled if it was missing?

Unless we have a way to test this, I suggest to keep it as is and as a follow up suggest to drop the check with a clear reasoning.

Reply:
The commit supposed to add it, and not it doesnt so it didnt seem right to mention it was dropped.

> I am having hard time understanding if this is indeed redundant or not. I do not see how it is assured that the resource has been assigned.
> 
> Are you saying that the pod was not suppose to be scheduled if it was missing?

Kubevirt add the resource request (through the pod `spec.resource.requests`), if the resource is not available the pod wont start.

> Unless we have a way to test this, I suggest to keep it as is and as a follow up suggest to drop the check with a clear reasoning.

I am afraid there is no way to test it because we need a real k8s API server to make the pod fail if /dev/vhost-net is not available.

Reply:
So how about doing think only in a follow up and ask others to confirm this is indeed correct.

Reply:
Following offline discussion, we saw its possible and decided to drop the check that exist in the Kubevirt core implementation https://github.com/kubevirt/kubevirt/pull/10542